packages/editor/cypress/integration/scriptlab_spec.js

```diff
@@ -0,0 +1,280 @@
+      {},
+    );
+
+    const startingSolutionCount = solutions.reduce((count, solution) => {
+      return count + (solution.host === host ? 1 : 0);
+    }, 0);
+
+    openEditorWithState({
+      sessionStorage: { host },
+      localStorage,
+    });
+
+    const solutionToDelete = ReportGeneration;
+    openSolutionFromEditor(solutionToDelete.id);
+
+    // click delete button
+    getByTestId('delete').click();
+    getByTestId('no-button').click(); // should close dialog
+
+    getByTestId('delete').click();
+    getByTestId('yes-button').click(); // this time click yes
+
+    // open backstage and validate that there is an additional solution
+    getByTestId('nav').click();
+    getByTestId('my-solution-list')
+      .children()
+      .should('have.length', startingSolutionCount - 1);
+  });
+
+  it('change tabs (switch files)', () => {
+    // opening with empty localStorage should land on backstage
+    openEditorWithState({
+      sessionStorage: { host: 'EXCEL' },
+    });
+
+    // click new on backstage
+    getByTestId('new').click();
+
+    // verify that CSS isn't selected
+    getByTestId('file-switcher-pivot')
+      .contains('CSS')
+      .as('CSS')
+      .should('not.be', 'active')
+      .click();
+
+    // verify that CSS is selected
+    cy.get('@CSS').should('be', 'active');
+  });
+
+  it('edit solution name and description', () => {
+    const solutions = [Solution1, BlankExcel, ReportGeneration];
+    const host = 'EXCEL';
+    const localStorage = solutions.reduce(
+      (all, solution) => ({ ...all, [`solution#${solution.id}`]: solution }),
+      {},
+    );
+
+    openEditorWithState({ sessionStorage: { host }, localStorage });
+
+    openSolutionFromEditor(ReportGeneration.id);
+
+    const newSolutionInfo = {
+      name: 'New Solution Name',
+      description: 'New Description',
+    };
+
+    getByTestId('solution-name').click();
+
+    getByTestId('solution-name-field')
+      .clear()
+      .type(newSolutionInfo.name);
+    getByTestId('solution-desc-field')
+      .clear()
+      .type(newSolutionInfo.description);
+
+    cy.wait(100);
+    getByTestId('update').click();
+
+    // info should be updated
+
+    getByTestId('solution-name').contains(newSolutionInfo.name);
+
+    getByTestId('nav').click();
+
+    getByTestId('my-solution-list').contains(newSolutionInfo.name);
+    getByTestId('my-solution-list').contains(newSolutionInfo.description);
+  });
+
+  it('edit a solution (in memory)', () => {
+    const solutions = [Solution1, BlankExcel, ReportGeneration];
+    const host = 'EXCEL';
+    const localStorage = solutions.reduce(
+      (all, solution) => ({ ...all, [`solution#${solution.id}`]: solution }),
+      {},
+    );
+
+    openEditorWithState({
+      sessionStorage: { host },
+      localStorage,
+    });
+
+    openSolutionFromEditor(ReportGeneration.id);
+
+    const textToSet = '// This is a test\n ';
+    clearEditor()
```

Comment:
I think you should stick some "awaits" in here...

Reply:
Cypress chainables, while they look like promises kind of, are not promises, and so awaiting them will not work.